tiling.js

```diff
@@ -1367,6 +1367,15 @@ class Spaces extends Map {
                 this.removeSpace(space);
             }
         }
+
+        // Ensure the space settings use correct indices
+        for (let [workspace, space] of this) {
+            space.settings.set_int('index', workspace.index());
+            // Trim to correct size too?
+            Meta.prefs_change_workspace_name(workspace.index(), space.name);
+            Settings.disconnectWorkspaceSettings(space.settings.get_str('uuid'));
```

Comment:
The disconnectWorkspaceSettings call should be in the "removed" loop above

Reply:
Might make sense to set the index to the first free index instead. So if there's there's three workspaces and we remove the second, we simply juggle the indexes like this:  [1, 2, 3] -> [1, 3, 2]. That way the preference isn't lost, but will be reused when a new workspace is added. To make the UX good we'd need some method to flip through previously used preferences though ofc., but at least we won't destroy stuff if someone tries out dynamic workspaces for a bit.